testsuite/pytests/test_getnodes.py

```diff
@@ -23,69 +23,74 @@
 Test GetNodes
 """
 
-import unittest
-
 import nest
-
-
-@nest.ll_api.check_stack
-class GetNodesTestCase(unittest.TestCase):
-    """Test GetNodes function"""
-
-    def setUp(self):
-        nest.ResetKernel()
-
-        a = nest.Create("iaf_psc_alpha", 3)  # noqa: F841
-        b = nest.Create("iaf_psc_delta", 2, {"V_m": -77.0})  # noqa: F841
-        c = nest.Create(
-            "iaf_psc_alpha", 4, {"V_m": [-77.0, -66.0, -77.0, -66.0], "tau_m": [10.0, 11.0, 12.0, 13.0]}  # noqa: F841
-        )
-        d = nest.Create("iaf_psc_exp", 4)  # noqa: F841
-
-    def test_GetNodes(self):
-        """test GetNodes"""
-        all_nodes_ref = nest.NodeCollection(list(range(1, nest.network_size + 1)))
-        all_nodes = nest.GetNodes()
-
-        self.assertEqual(all_nodes_ref, all_nodes)
-
-    def test_GetNodes_with_params(self):
-        """test GetNodes with params"""
-        nodes_Vm = nest.GetNodes({"V_m": -77.0})
-        nodes_Vm_ref = nest.NodeCollection([4, 5, 6, 8])
-
-        self.assertEqual(nodes_Vm_ref, nodes_Vm)
-
-        nodes_Vm_tau = nest.GetNodes({"V_m": -77.0, "tau_m": 12.0})
-        nodes_Vm_tau_ref = nest.NodeCollection([8])
-
-        self.assertEqual(nodes_Vm_tau_ref, nodes_Vm_tau)
-
-        nodes_exp = nest.GetNodes({"model": "iaf_psc_exp"})
-        nodes_exp_ref = nest.NodeCollection([10, 11, 12, 13])
-
-        self.assertEqual(nodes_exp_ref, nodes_exp)
-
-    def test_GetNodes_no_match(self):
+import pytest
+
+
+# Apply parameterization over number of threads here, each test will be run once for
+# each parameter value. Need to protect in case we are running without threads.
+@pytest.fixture(autouse=True, params=[1, 2] if nest.ll_api.sli_func("is_threaded") else [1])
+def create_neurons(request):
+    nest.ResetKernel()
+    nest.local_num_threads = request.param
+
+    nest.Create("iaf_psc_alpha", 3)
+    nest.Create("iaf_psc_delta", 2, {"V_m": -77.0})
+    # Work around because Create() with arrays does not work with more than one rank
+    # nest.Create("iaf_psc_alpha", 4, {"V_m": [-77.0, -66.0, -77.0, -66.0], "tau_m": [10.0, 11.0, 12.0, 13.0]})
+    c = nest.Create("iaf_psc_alpha", 4, {"V_m": -77.0, "tau_m": 10.0})
+    c[1].V_m = -66.0
+    c[3].V_m = -66.0
+    c[1].tau_m = 11.0
+    c[2].tau_m = 12.0
+    c[3].tau_m = 13.0
```

Comment:
I am curious how did this work, a local node (proxy-node) doesn't allow calling `set_status`, when working with`mpi`.

Reply:
I would try this example in python, because the python has a different implementation, so maybe `sli` drops the update for `non-local` nodes

Reply:
Yeah, It works, I just tested it too, we simply get a `nullptr` node when querying on a proxy-node, and we do nothing, on the other hand, PyNEST does not seem to check that when working with list of nodes (a.k.a `nodeCollection` with more than one item [issue: #3588]